Toolset/libraries/revbackscriptlibrary.livecodescript

```diff
@@ -2309,42 +2245,105 @@ private function revSEStackIsScriptEditor pStackname
-      lock messages
-      show stack tHiddenStack
-      if the mode of stack tHiddenStack is 0 then--Unix Window Managers 
-         if word 1 of tHiddenStack is "revPropertyPalette" then 
-            send "revUpdateTabs" to stack tHiddenStack
+      
+      local tHiddenStack
+      repeat for each key tHiddenStack in sHiddenPalettes
+         # OK-2009-10-08 : Its possible that the stack could be been removed between showing and hiding the palettes,
+         # this is due to extra mutable windows added with the new script editor. Check this first or it will keep throwing
+         # errors.
+         if there is no stack tHiddenStack then
+            next repeat
          end if
+         show stack tHiddenStack
+         if the mode of stack tHiddenStack is 0 then--Unix Window Managers 
+            if word 1 of tHiddenStack is "revPropertyPalette" then 
+               send "revUpdateTabs" to stack tHiddenStack
+            end if
+         end if
+      end repeat
+      
+      delete variable sHiddenPalettes
+   else
+      if the mode of stack "revErrorDisplay" is not 0 and revIDEGetPreference("cSEHideErrors") then
+         hide stack "revErrorDisplay"
+         put empty into sHiddenPalettes["revErrorDisplay"]
       end if
-   end repeat
-   put empty into sHiddenPalettes
+      
+      if the cSEHidePalettes of stack "revPreferences" then
+         local tOpenStacks
+         put the openStacks into tOpenStacks
+         repeat for each line tStackName in tOpenStacks
+            if not revIDEStackNameIsIDEStack(tStackName) then
+               next repeat
+            end if
+            
+            if not the visible of stack tStackName then
+               next repeat
+            end if
+            
+            if (tStackName is not among the items of kDontHide) and the mode of stack tStackName is 4 then
+               # OK-2008-06-24 : Bug 6621 : Prevent palettes being hidden if they are owned by the script editor
+               if the mainstack of stack tStackName is the short name of revScriptEditorMain() then
+                  next repeat
+               end if
+               
+               # OK-2009-12-22: Bug 8497 : Also keep separate variable viewer panes open
+               if tStackName begins with "revVariableVisualizer" then
+                  next repeat
+               end if
+               
+               hide stack tStackName
+               put empty into sHiddenPalettes[tStackName]
+            end if
+         end repeat
+      end if
+      -- must be after the others
+      if the mode of stack "Message Box" is not 0 and revIDEGetPreference("cSEHideMessageBox") then
+         hide stack "Message Box"
+         put empty into sHiddenPalettes["Message Box"]
+      end if
+   end if
+   
+   set the lockMessages to tOldLock
    unlock screen
-   unlock messages
-end revSEDelayShowPalettes
+end revSEDelayTogglePalettes
+
+
+private command log pMessage
+   put pMessage & return after msg
+end log
```

Comment:
@peter-b It looks like it can be factored out so I will do that... I'll add another commit to this patch that does further cleanup